src/main/java/net/ripe/rpki/rsyncit/RsyncitApplication.java

```diff
@@ -23,11 +35,39 @@ public static void main(String[] args) {
     }
 
     @Bean
-    public WebClientBuilderFactory webclientConfigurer(WebClient.Builder baseBuilder, AppConfig appConfig) {
-        // Explicit event loop is required for custom DnsNameResolverBuilder
-        NioEventLoopGroup group = new NioEventLoopGroup(1);
+    public WebClient.Builder webclientConfiguration(WebClient.Builder baseBuilder, AppConfig appConfig) {
+        final var userAgent = "rsyncit %s".formatted(appConfig.getInfo().gitCommitId());
 
-        return new WebClientBuilderFactory(group, baseBuilder, "rsyncit %s".formatted(appConfig.getInfo().gitCommitId()));
+        var httpClient = HttpClient.create()
+                .option(ChannelOption.CONNECT_TIMEOUT_MILLIS, 5000)
+                .responseTimeout(Duration.ofMillis(5000))
+                // remember: read and write timeouts are per read, not for a request.
+                .doOnConnected(conn ->
+                        conn.addHandlerLast(new ReadTimeoutHandler(5000, TimeUnit.MILLISECONDS))
+                            .addHandlerLast(new WriteTimeoutHandler(5000, TimeUnit.MILLISECONDS))
+                ).resolver(spec ->
+                    spec.resolvedAddressTypes(ResolvedAddressTypes.IPV6_PREFERRED)
+                        .completeOncePreferredResolved(false));
+
+
+        return baseBuilder
+                .clientConnector(new ReactorClientHttpConnector(httpClient))
+                .defaultHeader(HttpHeaders.USER_AGENT, userAgent);
     }
 
+    /**
+     * Return an observation customiser that only differs in that it omits the URL.
+     * The hostname for the request is the clientName.
+     *
+     * The full URL is in a high cardinality value (which would be used by observability tools)
+     */
+    @Bean
+    public ClientRequestObservationConvention nonUriClientRequestObservationConvention() {
```

Comment:
That is to not have unique tag for every new snapshot URL, right?

Reply:
That is what it should do, yeah. Think this is the spring magic way to go